.eleventy.js

```diff
@@ -1,22 +1,38 @@
-const fs = require("fs");
 const path = require("path");
-const yamlFont = require("yaml-front-matter");
 
 const pluginSyntaxHighlight = require("@11ty/eleventy-plugin-syntaxhighlight");
+const helpers = require("./templateHelpers");
 
 module.exports = function(eleventyConfig) {
   eleventyConfig.addPlugin(pluginSyntaxHighlight);
   eleventyConfig.addPassthroughCopy("css");
 
   eleventyConfig.addShortcode("getTitleBySlug", function(inputPath, slug) {
-    // TODO(robdodson): This may be expensive and slow. See if there's a way
-    // to do this without hitting the disk, or at least do it async.
-    // Maybe this should be a macro
-    const file = fs.readFileSync(
+    return helpers.getTitleBySlug(
```

Comment:
I was wondering if another way to do this would be to make sure every article has the tag `post`. Then you might be able to loop through the `post` collection and find the matching article? This might also let you get rid of the `basePath` property.

Reply:
It's good idea with the collections. I'd like to keep the basePath though, as this allows us to cater for cases where guides have the same slug (e.g. accessibility/intro.md and security/intro.md).

Reply:
I wonder if we would be able to get that same information using the collection file's `inputPath`?
https://www.11ty.io/docs/collections/#collection-item-data-structure